lxd/endpoints/socket.go

```diff
@@ -0,0 +1,98 @@
+package endpoints
+
+import (
+	"fmt"
+	"net"
+	"os"
+
+	lxd "github.com/lxc/lxd/client"
+	"github.com/lxc/lxd/shared"
+	"github.com/lxc/lxd/shared/logger"
+)
+
+// Bind to the given unix socket path.
+func socketUnixListen(path string) (net.Listener, error) {
+	addr, err := net.ResolveUnixAddr("unix", path)
+	if err != nil {
+		return nil, fmt.Errorf("cannot resolve socket address: %v", err)
+	}
+
+	listener, err := net.ListenUnix("unix", addr)
+	if err != nil {
+		return nil, fmt.Errorf("cannot bind socket: %v", err)
+	}
+
+	return listener, err
+
+}
+
+// Check if the socket at the given path is already bound to a running LXD
+// process, and return an error if so.
+//
+// FIXME: We should probably rather just try a regular unix socket
+//        connection without using the client. However this is the way
+//        this logic has historically behaved, so let's keep it like it
+//        was.
```

Comment:
Yeah, I wouldn't mind changing that, this should make things slightly faster too.

Reply:
Considering that a) the common case is that the socket file is not there at all and b) if it's there it should be stale and not associated with a process, there is not meaningful gain in performance to be made here. I'll keep it like this for now just to reduce risk, we could clean it up in a second round.